pages/garden/index.js

```diff
@@ -242,14 +264,43 @@ const ListPageLink = styled.div`
   color: var(--text-color);
 `;
 
-function Garden({
-  petCollection,
+export default function Garden({
+  activePet,
   setPetCollection,
   onInteractPet,
   currentPet,
   onCurrentPet,
 }) {
   const [animationState, setAnimationState] = useState(null);
+  const [characteristicEffects, setCharacteristicEffects] = useState(() => {
+    const speedFactor = activePet.characteristics.includes("hyperactive")
+      ? 0.4
+      : activePet.characteristics.includes("lethargic")
+      ? 2
+      : 1;
+    const happinessFactor =
+      (activePet.characteristics.includes("cheerful")
+        ? 0.5
+        : activePet.characteristics.includes("melancholy")
+        ? 1.5
+        : 1) *
+      (activePet.characteristics.includes("foolish")
+        ? 0.5
+        : activePet.characteristics.includes("smart")
+        ? 1.5
+        : 1);
+    const hungerFactor = activePet.characteristics.includes("gluttonous")
+      ? 1.5
+      : activePet.characteristics.includes("temperate")
+      ? 0.5
+      : 1;
+
+    return {
+      speedFactor: speedFactor,
+      happinessFactor: happinessFactor,
+      hungerFactor: hungerFactor,
+    };
+  });
```

Comment:
I find it really hard to read this section of your code - especially the `happinessFactor` and propose introducing a function that calculates the factor:

```suggestion
  function getHappinessFactor(characteristics) {
    const moodFactor = characteristics.includes("cheerful")
      ? 0.5
      : characteristics.includes("melancholy")
      ? 1.5
      : 1;

    const intelligenceFactor = characteristics.includes("foolish")
      ? 0.5
      : characteristics.includes("smart")
      ? 1.5
      : 1;

    return moodFactor * intelligenceFactor;
  }

  const happinessFactor = getHappinessFactor(activePet.characteristics);
});
```

This will mean more code, but it makes it easier to follow at a quick glance what is happening.
You could even consider calculating all factors in separate functions.